tools/core-build-tasks/src/tasks/bundle.ts

```diff
@@ -17,20 +21,124 @@ export type BundleTaskParameters = {
     /** The output file for the bundle. Documentation: https://esbuild.github.io/api/#outfile */
     outfile: string;
 
-    /** Flag to specify how to generate source map. Documentation: https://esbuild.github.io/api/#sourcemap */
+    /** Flag to specify to generate a source map file. Documentation: https://esbuild.github.io/api/#sourcemap*/
     sourcemap?: boolean | 'linked' | 'inline' | 'external' | 'both';
+
+    /** The output path for the source map file. Ignored if sourcemap is false or 'inline'. */
+    outputSourcemapPath?: string;
+};
+
+export type PostProcessOutputFilesResult = {
+    sourceMapDirectory: string;
+    outputDirectory: string;
+    generatedFiles: Record<string, string>;
 };
 
+function isRequiredToMakeAnyFileChange(
+    sourcemap: boolean | 'linked' | 'inline' | 'external' | 'both' | undefined
+): boolean {
+    return sourcemap === true || sourcemap === 'linked' || sourcemap === 'external' || sourcemap === 'both';
```

Comment:
super nit: the negative condition is probably more concise 😄 

If this is my only feedback though, feel free to ignore

Reply:
Done